llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -1490,13 +1486,29 @@ class LoopVectorizationCostModel {
     }
   }
 
+  /// Disables previously chosen tail folding policy, sets it to None. Expects,
+  /// that the tail policy was selected.
+  void disableTailFolding() {
+    assert(ChosenTailFoldingStyle && "Tail folding must be selected.");
+    ChosenTailFoldingStyle =
+        std::make_pair(TailFoldingStyle::None, TailFoldingStyle::None);
+  }
+
```

Comment:
Maybe, will try to do

Reply:
After some thoughts, I think this may not be possible for non-power-of-2 distance. Here we need to be sure that tail-folding-with-EVL is chosen to enable selection of the non-power-of-2 maxsafedist